### ISSUE TITLE
False positive for DMI_RANDOM_USED_ONLY_ONCE with Stream methods

### ISSUE THREAD

**Issue by md5** [spotbugs/spotbugs#2370]
I'm getting a false positive on `DMI_RANDOM_USED_ONLY_ONCE` for code like the following:

```java
import java.util.Random;
import java.util.stream.IntStream;

public class RandomTest {
  public IntStream randomPercents(int count) {
    return new Random().ints(count, 0, 100);
  }
}
```

Returns this error:

```
H B DMI: Random object created and used only once in RandomTest.randomPercents(int)  At RandomTest.java:[line 6]
```

Since `ints()` is returning a stream, using that method or any of the analogous methods like `longs()` should act as if the `Random` used more than once. I suppose there's an edge case of `count` is `0` or `1`.

**Comment by ThrawnCA** [comment]
The `ints` method does look like a good candidate for being treated as "yes this Random is being reused".
